### PR TITLE
ML-038/add error and status handling

### PR DESCRIPTION
**Pull request by WhatWasThatt**
# Description

update rubocop.yml only without lib/tasks directory
add error handling module
update application.rb with error handling module
include error handling module to the application controller
fix aasm error
fix protect_from_forgery error

## Trello card link

https://trello.com/c/oO0vZl6C

# Checklist:

- [x] My code fixes the problem or adds required functionality
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [ ] I have added tests that prove that my feature works
- [x] New and existing unit tests pass locally with my changes

# Additional requirements:

- [ ] Does your code require any migrations to be ran?
- [x] Does your code require installing any new gems or system dependencies?
- [ ] Does your code require any changes to current deployment process?

# Additional Information

### CODE REVIEW

File: lib/error/error_handler.rb [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+          respond(:standard_error, 500, e.to_s)
+        end
+      end
+    end
+
+    private
+
+    def respond(error, status, message)
+      json = Helpers::Render.json(error, status, message)
+      render json: json, status: status
```
Comment (twill11) [inline lib/error/error_handler.rb:95]: GJ! but i think u need a empty line under 94, what do you think?
image

File: lib/error/error_handler.rb [hunk 2]
```diff
@@ -0,0 +1,98 @@
+module Error
+  module ErrorHandler
+    def self.included(clazz)
+      clazz.class_eval do
+        rescue_from ActionController::RoutingError do |error|
+          respond(:routing_error, 404, error.to_s)
+        end
```
Comment (twill11) [inline lib/error/error_handler.rb:7]: can it be in this block? { }

File: lib/error/error_handler.rb (hunk 1 above)
Comment (WhatWasThatt) [reply lib/error/error_handler.rb:95]: it's not an endpoint, it's a function

File: lib/error/error_handler.rb (hunk 2 above)
Comment (WhatWasThatt) [reply lib/error/error_handler.rb:7]: we have a rubokop rule about the length of the line

**Review by twill11 — APPROVED** [review]
nice!

**Review by KarinaKlevzhits — APPROVED** [review]
for me looks good

**Review by KarinaKlevzhits — APPROVED** [review]

**Review by dmitrymalllpr — CHANGES_REQUESTED** [review]
https://medium.com/rails-ember-beyond/error-handling-in-rails-the-modular-way-9afcddd2fe1b
https://medium.com/ruby-on-rails-web-application-development/custom-400-500-error-pages-in-ruby-on-rails-exception-handler-3a04975e4677
Need rework

**Review by twill11 — APPROVED** [review]
gj!

**Review by dmitrymalllpr — APPROVED** [review]

**Pull request merged**